src/DynamoCoreWpf/Views/Core/DynamoView.xaml

```diff
@@ -641,7 +641,8 @@
                     <MenuItem Header="{x:Static p:Resources.DynamoViewSettingMenu}"
                               Name="SettingsMenu"
                               Focusable="False"
-                              IsEnabled="True">
+                              IsEnabled="True"
+                              SubmenuOpened="onSettingsSubMenuOpened">
```

Comment:
Since the isEnabled property of `Geometry Scaling` menu item should be updated before the setting menus is opened, I added the `SubmenuOpened` to the upper menu item, which is `Settings`.